pennylane/estimator/compact_hamiltonian.py

```diff
@@ -178,3 +180,307 @@ def __post_init__(self):
 
         if isinstance(self.one_norm, int):
             object.__setattr__(self, "one_norm", float(self.one_norm))
+
+
+class PauliHamiltonian:
+    r"""Stores the minimum necessary information required for resource estimation of a
+    Hamiltonian expressed as a linear combination of tensor products of Pauli operators.
+
+    Args:
+        num_qubits (int): total number of qubits the Hamiltonian acts on
+        num_pauli_words (int | None): the number of terms (Pauli words) in the Hamiltonian
+        max_weight (int | None): The maximum number of qubits a term acts upon, over all
+            terms in the linear combination.
+        one_norm (float | int | None): the one-norm of the Hamiltonian
+        pauli_dist (dict | None): A dictionary representing the various Pauli words and how
+            frequently they appear in the Hamiltonian.
+        commuting_groups (tuple(dict) | None): A tuple of dictionaries where each entry represents
+            a group of Pauli words that mutually commute. Each entry is formatted similarly to the
+            ``pauli_dist`` argument (see the Usage Details section for more information).
+            of terms from the Hamiltonian such that all terms in the group commute. Here each
+            dictionary contains the various Pauli words and how frequently they appear in the group.
+
+    Returns:
+        PauliHamiltonian: An instance of PauliHamiltonian
+
+    .. seealso::
+        :class:`~.estimator.templates.TrotterPauli`, :class:`~.estimator.templates.SelectPauli`
+
+    **Example**
+
+    A ``PauliHamiltonian`` is a compact representation which can be used with compatible templates
+    to obtain resource estimates. Consider for example the Hamiltonian:
+
+    .. math::
+
+        \hat{H} = 0.1 \cdot \Sigma^{30}_{j=1} \hat{X}_{j} \hat{X}_{j+1}
+        - 0.05 \cdot \Sigma^{30}_{k=1} \hat{Y}_{k} \hat{Y}_{k+1} + 0.25 \cdot \Sigma^{40}_{l=1} \hat{X}_{l}
+
```

Comment:
Please add a sentence here saying: that this Hamiltonian can be represent in a compact way with PauliHamiltonian as

Reply:
Added 👍🏼